pallets/identity/src/lib.rs

```diff
@@ -267,6 +273,15 @@ decl_module! {
         fn deposit_event() = default;
 
         fn on_runtime_upgrade() -> Weight {
+            use frame_support::migration::{StorageIterator, put_storage_value};
+            // Covert old scopes to new scopes
+            for (key, value) in StorageIterator::<Claim2ndKeyOld>::new(b"Identity", b"Claims").drain() {
+                let new = Claim2ndKey {
+                    issuer: value.issuer,
+                    scope: value.scope.map(|scope| Scope::Identity(scope)),
+                };
+                put_storage_value(b"Identity", b"Claims", &key, new);
```

Comment:
This won't work because `Claim2ndKeyOld` is part of the key. It's not the value.

One way to do the migration will be to use `IterableStorageDoubleMap` to iterate over all key1, key2, values and insert key1, new_key2, value in a different map. The only problem here is that It'll require the new mapping name to be different.

Consider asking the parity folks if they have something like `StorageIterator` for double maps where you need to migrate one of the keys, not the value. Implementing one from scratch will be a tedious task and I'd rather just use the above method and live with a new mapping name.